### ISSUE TITLE
LocationLayer plugin does not behave properly with plugin version 0.3.0 and Mapbox Android SDK 5.3.0

### ISSUE THREAD

**Issue by Tweener** [mapbox/mapbox-plugins-android#250]
Hi,
I'm facing a problem with the LocationLayer plugin since I upgraded its version to 0.3.0 and MapBox Android SDK to 5.3.0.

I was using Mapbox Android SDK version 5.1.0 and with LocationLayer plugin version 0.2.0 and I was able to receive location updates properly. I could see the blue dot pointing exactly to my current location. And if I moved, the blue dot would move as well. Everything worked as expected. 

This was the Gradle configuration:
```
implementation('com.mapbox.mapboxsdk:mapbox-android-sdk:5.1.0@aar') {
    transitive = true
}
implementation 'com.mapbox.mapboxsdk:mapbox-android-plugin-locationlayer:0.2.0'
```

I had one small problem, related to markers blinking randomly (as explained in this issue https://github.com/mapbox/mapbox-gl-native/issues/8799), which has been fixed since then. That's why I decided to upgrade the versions.

This is now the updated Gradle configuration:
```
implementation 'com.mapzen.android:lost:3.0.2'
implementation('com.mapbox.mapboxsdk:mapbox-android-sdk:5.3.0@aar') {
    transitive = true
}
implementation 'com.mapbox.mapboxsdk:mapbox-android-plugin-locationlayer:0.3.0'
```
(I've read [in the 5.2.0 release changelog](https://github.com/mapbox/mapbox-gl-native/releases/tag/android-v5.2.0) that we need to add the LOST dependency.)

What happens now is that the blue dot is set to a location I'm not at and stays there. Or sometimes starts at that wrong location and moves to right location, without the camera moving. (As shown in the GIFs below).
The thing is, it used to work properly before changing versions. The camera and the blue dot were set directly to the right location, without moving from a wrong location the right one on `onMapReady()`.

I didn't find a proper sample showing how to update to MapboxAndroidSDK 5.3.0 and LocationLayer plugin 0.3.0, as the actual samples seem outdated. 

Is this a known issue? Am I doing something wrong?

Thanks.

Pointing to a wrong location:
![mapbox_location_bug1](https://user-images.githubusercontent.com/596985/35307289-60174fde-00a2-11e8-896e-831c26495e9d.gif)

Starting from a wrong location then moving to the right location, but the camera won't move:
![mapbox_location_bug](https://user-images.githubusercontent.com/596985/35306855-314c1eca-00a0-11e8-9a42-6084f344742f.gif)

**Comment by Tweener** [comment]
Up. Any idea?

**Comment by daesu** [comment]
Experiencing the same issue; 

com.mapzen.android:lost:3.0.4
com.mapbox.mapboxsdk:mapbox-android-sdk:5.4.0
com.mapbox.mapboxsdk:mapbox-android-plugin-locationlayer:0.3.0

**Comment by cammace** [comment]
0.3.0 location layer doesn't offer any camera tracking (#259 will add this logic). I tested on 0.4.0-SNAPSHOT and wasn't able to reproduce the location starting off in the wrong location. Any chance you could test out the snapshot and let me know if you still see the issue?

**Comment by cammace** [comment]
0.5.0-beta.2's available and the final release is close to happening. We also just released 6.0.1 of the map sdk which includes significant changes. Since this issue is pretty stale, i'm going to go ahead and close but if the issue persist even after updating, I'll reopen.
